### PR TITLE
Added tests for Login with spam protection

### PR DESCRIPTION
**Pull request by dhlavac**

### CODE REVIEW

File: testsuite/ui/views/devel/login.py [hunk 1]
```diff
@@ -48,6 +65,98 @@ def do_rhsso_login(self, username, password):
...
     @property
-    def is_displayed(self, ):
+    def is_displayed(self):
         return self.path_pattern in self.browser.url and self.username_field.is_displayed and \
                self.password_field.is_displayed and self.sign_in_btn.is_displayed
+
+
+class ReCaptcha(View):
+    """View that represents the reCAPTCHA box"""
+    FRAME = "//iframe[@title='reCAPTCHA']"
```
Comment (JaurbanRH) [inline testsuite/ui/views/devel/login.py:75]: Are you sure that this need to be here? If yes please add comment why.

File: testsuite/ui/views/devel/login.py [hunk 2]
```diff
@@ -48,6 +65,101 @@ def do_rhsso_login(self, username, password):
...
+    path_pattern = "/signup/success"
+    thank_you = Text("//div[contains(@class, 'panel-body panel-footer')]/h2")
+
+    @property
+    def is_displayed(self):
+        return BaseDevelView.is_displayed.fget(self) and self.path in self.browser.url \
+               and self.thank_you.text == "Thank you"
+
+
+class DevelForgotPassword(BaseDevelView):
```
Comment (jsmolar) [inline testsuite/ui/views/devel/login.py:137]: Naming: "ForgotPasswordView"

File: testsuite/ui/views/devel/login.py [hunk 3]
```diff
@@ -48,6 +65,101 @@ def do_rhsso_login(self, username, password):
...
+    def prerequisite(self):
+        return LoginDevelView
+
+    @property
+    def is_displayed(self):
+        return SignUpView.is_displayed.fget(self) and self.path in self.browser.url and \
+               self.password.is_displayed and self.password2.is_displayed
+
+
+class SuccessfulAccountCreation(BaseDevelView):
```
[the remark this answers is not in the document]
Comment (dhlavac) [reply testsuite/ui/views/devel/login.py:126]: There is different endpoint in URL on successful page

File: testsuite/tests/ui/devel/auth/test_login_recaptcha.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+    yield user_name
+
+    if not testconfig["skip_cleanup"]:
+        usr = threescale.accounts.read_by_name(user_name)
+
+        def finalizer():
+            usr.delete()
+
+        request.addfinalizer(finalizer)
```
Comment (jsmolar) [inline testsuite/tests/ui/devel/auth/test_login_recaptcha.py:26]: use lambda here

Comment (jsmolar) [inline testsuite/tests/ui/devel/auth/test_login_recaptcha.py:26]: use labda here

File: testsuite/tests/ui/devel/auth/test_login_recaptcha.py [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    """
+    Opens the developer portal and passes in the access code.
+    @param browser: Browser instance
+    @param navigator: Navigator Instance
+    @param provider_account: Currently used provider account (tenant)
+    """
+    url = settings["threescale"]["devel"]["url"]
+    browser.url = url
+    browser.selenium.delete_all_cookies()
+    navigator.open(LoginView, access_code=provider_account['site_access_code'])
```
Comment (jsmolar) [inline testsuite/tests/ui/devel/auth/test_login_recaptcha.py:41]: `navigator.open(LoginView, url=url, access_code=provider_account['site_access_code'])`

File: testsuite/tests/ui/devel/auth/test_login_recaptcha.py [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+# pylint: disable=too-many-arguments
+@pytest.fixture(scope="module")
+def spam_protection_setup(custom_admin_login, browser, navigator):
+    """
+    Opens the developer portal and passes in the access code.
+    @param browser: Browser instance
+    @param navigator: Navigator Instance
+    @param custom_admin_login: Currently used provider account (tenant)
+    """
+    custom_admin_login()
```
Comment (jsmolar) [inline testsuite/tests/ui/devel/auth/test_login_recaptcha.py:53]: `login` should be enough

File: testsuite/tests/ui/devel/auth/test_login_recaptcha.py [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+    spam_page = navigator.open(SpamProtection)
+    assert spam_page.sus_protection.is_enabled, "Missing recaptcha setup in 3scale"
+    assert spam_page.always_protection.is_enabled, "Missing recaptcha setup in 3scale"
+    spam_page.enable_always_protection()
+
+    yield
+
+    url = settings["threescale"]["admin"]["url"]
+    browser.url = url
+    page = navigator.open(SpamProtection)
```
Comment (jsmolar) [inline testsuite/tests/ui/devel/auth/test_login_recaptcha.py:63]: `navigator.open(SpamProtection, url=url)`

**Review by jsmolar — APPROVED** [review]

**Pull request merged**